flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestServerEndpoint.java

```diff
@@ -113,6 +115,15 @@ public RestServerEndpoint(RestServerEndpointConfiguration configuration) throws
 		this.responseHeaders = configuration.getResponseHeaders();
 
 		terminationFuture = new CompletableFuture<>();
+
+		// the clean up of temp files when jvm exits is handled by org.apache.flink.util.ShutdownHookUtil; thus,
+		// it's no need to register those files (post chunks and upload file chunks) to java.io.DeleteOnExitHook
+		// which may lead to memory leak.
+		DiskAttribute.deleteOnExitTemporaryFile = false;
+		DiskFileUpload.deleteOnExitTemporaryFile = false;
+
+		// share the same directory with file upload for post chunks storage.
+		DiskAttribute.baseDirectory = uploadDir.normalize().toAbsolutePath().toString();
```

Comment:
Is the `DiskAttribute` and `DiskFileUpload` only used by Flink's `FileUploadHandler`? If yes, would it make sense to move these settings into the constructor of the `FileUploadHandler`? At the moment we also set the `DiskFileUpload.baseDirectory` there. That way it would be easier to understand the context.

Reply:
You are right. The DiskAttribute and DiskFileUpload is only used by FileUploadHandler. I will move these settings into the constructor of the FileUploadHandler and update the pr.

Reply:
I have updated the pr.